modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Fields.tsx

```diff
@@ -749,6 +749,8 @@ const Fields = ({
 								(fdsField: IFDSField) => fdsField.id !== item.id
 							) || []
 						);
+
+						updateFDSFieldsOrder();
```

Comment:
Just so I understand this, are you expecting that by adding this function here that it will update the order to no longer include the FDS field that you are deleting?

If so, I don't think that will work because the new fdsFields hasn't been set yet. I believe `updateFDSFieldsOrder` would require re-render of the component first(after setting the fields in line 747) because update is tied to a ref that is only gonna be updated when this component re-renders.

Reply:
Can you give me a code example, please? I understand your point but I'm not sure how to "re-render the component before calling `updateFDSFieldsOrder`"

Reply:
I don't have a specific example of code, but just thinking through the flow of data here...

When you delete an item, you are calling "setState" somewhere(either in this component or a parent component). Now we are still in this same function, so we can't be sure that the state is set yet. Now when we look at `updateFDSFieldsOrder`, we see that it is getting the data from a ref(`fdsFieldsOrderRef.current`). Now that should raise some questions because now we are depending on two different means of data, one via state(which is tied to the react lifecycle) and one via ref(which is a global value not related to lifecycle).

If deleting an item is dependent on the lifecycle and updating the order is not related to the lifecycle, that is an indicator that this probably wouldn't work as expected.

My recommendation would be to update `updateFDSFieldsOrder` so that it also allows for the ability to accept an array of items, instead of solely depending on the ref. Or the other option is to manually set that ref right before calling `updateFDSFieldsOrder`.

Reply:
I see, thanks for looking into it! The logic behind these actions (using 2 types of variables: state and a ref) was already there, does it mean it wasn't "ok" before my change?

In any case I did a test trying to unify everything and avoiding using variables (state or ref) but it wasn't working either https://github.com/liferay-frontend/liferay-portal/pull/3607/commits/a269ddca5d754e8820a0093b7eef8c6388e2b8c1 although maybe it's the same error Kevin mentioned in https://github.com/liferay-frontend/liferay-portal/pull/3595#issuecomment-1702020691

Reply:
It's not that is wasn't okay before, as I think the ref is necessary for drag and drop reordering. But when combining the two, it just takes an approach of thinking through what the lifecycle of the component will be and what data is actually being used for each async action